csharp/source/autowrap/csharp/common.d

```diff
@@ -8,15 +10,55 @@ public struct RootNamespace {
     string value;
 }
 
-package string getInterfaceTypeString(T)() {
-    import std.datetime : DateTime, SysTime;
-    import std.traits : fullyQualifiedName;
+public string generateSharedTypes() {
```

Comment:
Wouldn't this be better as a template mixin?

I'm also trying to understand why the user needs this to be mixed in to their code instead of library code that's called.

Reply:
Maybe i'm going a bit too crazy with mixins. In this case the datetime type is used in the AliasSeq on line 35 of dlang.d as well as being included in the generated boilerplate interface code.

Reply:
I only used mixins where needed, especially since pyd needs a lot boilerplate. AFAIK this might be needed, but I'm sceptical and would like to undetrstand why. Why does the `datetime` type have to be in the user's code at all?

Reply:
It's not in the user's code per se, it's in the generated interface code that gets mixed into the user's library. I actually do this with the return types as well. The technical reason is that because none of the Date/Time types in either language can be shared across the language boundaries there must exist some intermediate type that **can** be shared across the language boundary in order to properly marshal the Date/Time data.

Reply:
I understand the necessity of an intermediate type.

What I'm trying to understand is why `datetime` is created as a mixin instead of, say, `autowrap.csharp.datetime.datetime` as a regular type. That is: why isn't it regular library code instead of a mixin? 

Python requires a function with a particular name to initialise the library. I don't know of any other way to generate that for the user without using a mixin. I'm thinking now `DllMain` shouldn't have been a mixin but just regular library code as well, added as a dependency for autowrap:python and autowrap:excel.

Looking at the return type code, I don't get why the mixins are needed there either.

Reply:
I actually agree with you. I guess it's just easy to over use the latest cool tool you just learned about. :) Fixed. This ended up being a pretty significant refactoring.